### PR TITLE
[FLINK-29157][doc] Clarify the contract between CompletedCheckpointStore and SharedStateRegistry

### PR DESCRIPTION
**Pull request by fredia**
## What is the purpose of the change

Improve logging in CompletedCheckpointStore

## Verifying this change

This change is a trivial rework without any (new) test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature?  no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CompletedCheckpointStore.java [hunk 1]
```diff
@@ -39,6 +41,15 @@ public interface CompletedCheckpointStore {
      * <p>Only a bounded number of checkpoints is kept. When exceeding the maximum number of
      * retained checkpoints, the oldest one will be discarded.
      *
+     * <p>After <a href="https://issues.apache.org/jira/browse/FLINK-24611">FLINK-24611</a>, {@link
+     * SharedStateRegistry#unregisterUnusedState} should be called here to subsume unused state.
+     * <font color="#FF0000"><strong>Note</strong></font>, the {@link CompletedCheckpoint} passed to
+     * SharedStateRegistry must be the same object as the input parameter, otherwise the state may
```
Comment (masteryhx) [inline flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CompletedCheckpointStore.java:47]: IIUC, the fallible place is SharedStateRegistryFactory#create where the same object should be passed.
So maybe specify it here ?

**Comment by fredia** [comment]
Hi @klion26, could you please take a review?

**Review by klion26 — CHANGES_REQUESTED** [review]
Thanks for your contribution, please take a look at the comment inline.

**Review by klion26 — APPROVED** [review]
Thanks for the update, LGTM now, merging

**Comment by fredia** [comment]
Thanks a lot for reviewing. 👍

**Pull request merged**